### PR TITLE
[dartsim] Add new port

### LINKED ISSUE #10015: [New Port Request]  Dynamic Animation and Robotics Toolkit (DART)
Library name: Dynamic Animation and Robotics Toolkit (DART)

Library description: DART (Dynamic Animation and Robotics Toolkit) is a collaborative, cross-platform, open source library that provides data structures and algorithms for kinematic and dynamic applications in robotics and computer animation. 

Source repository URL: https://github.com/dartsim/dart

Project homepage (if different from the source repository): http://dartsim.github.io/

Anything else that is useful to know when adding (such as optional features the library may have that should be included):
* Repology: https://repology.org/project/dart/versions and https://repology.org/project/dartsim/versions
* Debian package: https://packages.debian.org/source/sid/dart
* Homebrew Formula: https://formulae.brew.sh/formula/dartsim
* Third-party port repository, ready to be imported: https://github.com/jslee02/vcpkg-dart

### PR DESCRIPTION
**Pull request by traversaro**
**Describe the pull request**

- What does your PR fix? This PR adds a new port for the DART (Dynamic Animation and Robotics Toolkit) library, a articulated body simulated library used in Robotics and Graphics (http://dartsim.github.io/). Fix https://github.com/microsoft/vcpkg/issues/10015 .

- Which triplets are supported/not supported? Have you updated the CI baseline?
  As far as I know all the triplet should be supported. 

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

### CODE REVIEW

**Comment by traversaro** [comment]
Apparently this need a bit more work, marking it as WIP.

**Comment by PhoebeHui** [comment]
The second patch failed to apply, could you update it?

Checking patch cmake/DARTMacros.cmake...
error: while searching for:
    endif()
  else()
    set(HAVE_${variable} FALSE CACHE BOOL "Check if ${variable} found." FORCE)
    if(ARGV3) # version
      message(WARNING "Looking for ${dependency} - NOT found, to use"
                      " ${component}, please install ${dependency} (>= ${ARGV3})")
    else()
      message(WARNING "Looking for ${dependency} - NOT found, to use"
                      " ${component}, please install ${dependency}")
    endif()
    return()
  endif()

error: patch failed: cmake/DARTMacros.cmake:135
error: cmake/DARTMacros.cmake: patch does not apply

**Comment by traversaro** [comment]
Yes, I already fixed it locally, I need to push it upstream, but the port is still blocked by https://github.com/microsoft/vcpkg/pull/12795 .

**Comment by traversaro** [comment]
> Yes, I already fixed it locally, I need to push it upstream, but the port is still blocked by #12795 .


https://github.com/microsoft/vcpkg/pull/12795 has been merged, I just rebased the PR on the top of the latest master.

**Comment by traversaro** [comment]
The PR is now failing due to the error: 
~~~
FAILED: dart/utils/CMakeFiles/dart-utils.dir/XmlHelpers.cpp.o 
/usr/bin/c++  -DBOOST_TEST_DYN_LINK -I/mnt/vcpkg-ci/buildtrees/dartsim/src/v6.9.4-8a922c6dbc.clean -I. -isystem /mnt/vcpkg-ci/installed/x64-linux/share/eigen3/../../include/eigen3 -isystem /mnt/vcpkg-ci/installed/x64-linux/include -fPIC -Wall -Wextra -fPIC -g -fno-omit-frame-pointer -fno-inline-functions -fno-inline-functions-called-once -fno-optimize-sibling-calls -MD -MT dart/utils/CMakeFiles/dart-utils.dir/XmlHelpers.cpp.o -MF dart/utils/CMakeFiles/dart-utils.dir/XmlHelpers.cpp.o.d -o dart/utils/CMakeFiles/dart-utils.dir/XmlHelpers.cpp.o -c /mnt/vcpkg-ci/buildtrees/dartsim/src/v6.9.4-8a922c6dbc.clean/dart/utils/XmlHelpers.cpp
In file included from /mnt/vcpkg-ci/installed/x64-linux/include/boost/lexical_cast/detail/converter_lexical_streams.hpp:63:0,
                 from /mnt/vcpkg-ci/installed/x64-linux/include/boost/lexical_cast/detail/converter_lexical.hpp:54,
                 from /mnt/vcpkg-ci/installed/x64-linux/include/boost/lexical_cast/try_lexical_convert.hpp:44,
                 from /mnt/vcpkg-ci/installed/x64-linux/include/boost/lexical_cast.hpp:32,
                 from /mnt/vcpkg-ci/buildtrees/dartsim/src/v6.9.4-8a922c6dbc.clean/dart/utils/XmlHelpers.cpp:38:
/mnt/vcpkg-ci/installed/x64-linux/include/boost/lexical_cast/detail/inf_nan.hpp:34:10: fatal error: boost/math/special_functions/sign.hpp: No such file or directory
 #include <boost/math/special_functions/sign.hpp>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~

That is probably due to https://github.com/microsoft/vcpkg/issues/8753 . However, an obvious workaround is to add an explicit dependency on `boost-math`.

**Comment by traversaro** [comment]
The CI tests are now running correctly.

**Review by PhoebeHui — APPROVED** [review]
The changes looks good to me, thanks for your contribution!

**Comment by ras0219-msft** [comment]
Thanks!

File: ports/dartsim/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "dartsim",
+  "version-string": "9.4.0",
```
Comment (jslee02) [inline ports/dartsim/vcpkg.json:3]: Shouldn't this version `6.9.4` to match the one in `portfile.cmake`?

Comment (traversaro) [reply ports/dartsim/vcpkg.json:3]: Good catch @jslee02, I am not sure how the 9.4.0 ended there, probably when I migrated from the `CONTROL` file to the new `vcpkg.json` I copied an existing file and I forgot to update the version. A fix for this is in https://github.com/microsoft/vcpkg/pull/13550 , let's hope the jump backword in version is not creating problems.

Comment (jslee02) [reply ports/dartsim/vcpkg.json:3]: Thanks for the quick patch!

**Pull request merged**